### PR TITLE
Frameworks

### PR DESCRIPTION
**Pull request by svnsairam**
### Description
Frameworks URL

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -173,6 +173,11 @@ export const serviceUsageOrigin = utils.envOverride(
   "FIREBASE_SERVICE_USAGE_URL",
   "https://serviceusage.googleapis.com"
 );
+// TODO: Change to correct URL later.
```
Comment (alexastrum) [inline src/api.ts:176]: Please assign an issue or a contributor to this TODO.

File: src/frameworks/compose/discover/frameworks.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+export type BuildOutputOnlyFields = "createTime" | "updateTime" | "sourceRef";
+
+interface BuildSource {
+  codeBaseSource?: CodebaseSource;
+}
+
+interface Status {
+  code: number;
+  message: string;
+  details: any[];
```
Comment (alexastrum) [inline src/frameworks/compose/discover/frameworks.ts:53]: Any chance we can have something more specific than `any` here?

File: src/api.ts (hunk 1 above)
Comment (svnsairam) [reply src/api.ts:176]: removing this - internal details.

File: src/frameworks/compose/discover/frameworks.ts (hunk 2 above)
Comment (svnsairam) [reply src/frameworks/compose/discover/frameworks.ts:53]: In the backend its type is declared as `Any` and I wasn't sure what that type looks like as it comes from backend.

File: src/api.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (svnsairam) [reply src/api.ts:176]: Done

File: src/frameworks/compose/discover/frameworkstackApi.ts
[the remark this answers is not in the document]
Comment (svnsairam) [reply src/frameworks/compose/discover/frameworkstackApi.ts:1]: Going forward do we use src/api/ for all api calls?

File: src/api/frameworks.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  statusDetail: string;
+  cancelRequested: boolean;
+  apiVersion: string;
+}
+
+export interface Operation {
+  name: string;
+  metadata?: OperationMetadata;
+  done: boolean;
+  // oneof result
```
Comment (taeold) [inline src/api/frameworks.ts:78]: q: what do these mean 🤔 ?

**Review by taeold — APPROVED** [review]

File: src/api/frameworks.ts (hunk 3 above)
Comment (svnsairam) [reply src/api/frameworks.ts:78]: These are part of Long Running Operation object and they are returned as response from the rpc method, 
name: Service assigned name which is unique. For HttpMapping, the `name` is a resource name ending with `operations/{unique_id}`.
OperationMetadata: Service specific metadata associated with the operation. The is the metadata_type of the Operation.
done: If `false` then Operation is still running, if `true` operation is completed.

**Pull request merged**